app/models/hub_edos/common/contact/email.rb

```diff
@@ -0,0 +1,57 @@
+module HubEdos
+  module Common
+    module Contact
+      class Email
+
+        # a short descriptor representing the kind email, such as work or personal etc.
+        attr_accessor :type
+
+        # the full email address, including account name, host name, and suffix	string
+        attr_accessor :address
+
+        # an indicator of whether this email address should be the main one used
+        attr_accessor :primary
+
+        # an indicator of whether this email address should be disclosed to the public
+        attr_accessor :disclose
+
+        # a short descriptor indicating limits on displaying or editing this email
+        attr_accessor_with_lower_camelized_alias :ui_control
+
+        # the date this email address came into use
+        attr_accessor_with_lower_camelized_alias :from_date
+
+        # the date this email address stopped being used
+        attr_accessor_with_lower_camelized_alias :to_date
+
+        alias :email_address :address
+        alias :email_address= :address=
+        alias :emailAddress :address
+        alias :emailAddress= :address=
+
+        def initialize(attrs={})
+          attrs.each do |key, value|
+            send("#{key}=", value) if respond_to?("#{key}=")
+          end
+          @type = HubEdos::Common::Reference::Descriptor.new(type) if type
+          @ui_control = HubEdos::Common::Reference::Descriptor.new(ui_control) if ui_control
+          @from_date = Date.parse(from_date) if from_date
+          @to_date = Date.parse(to_date) if to_date
+        end
+
+        def as_json(options={})
+          {
+            type: type.as_json,
+            address: address,
+            emailAddress: emailAddress,
+            primary: primary,
+            disclose: disclose,
+            uiControl: uiControl.as_json,
+            fromDate: (fromDate.present? ? fromDate.to_s : nil),
+            toDate: (toDate.present? ? toDate.to_s : nil),
```

Comment:
I would assume internally it would be preferable to use the underscored version of the methods here for clarity